Unable to access 0.0.0.master of com.regnosys.rosetta

User yzy108134@taifex.com.tw tried to execute a maven build on the code generators project and got the error
[ERROR] Failed to execute goal on project test-helper: Could not resolve dependencies for project com.regnosys.rosetta.code-generators:test-helper:jar:0.0.0.master: Failed to collect dependencies at com.regnosys.rosetta:com.regnosys.rosetta:jar:0.0.0.master: Failed to read artifact descriptor for com.regnosys.rosetta:com.regnosys.rosetta:jar:0.0.0.master: Could not transfer artifact com.regnosys.rosetta:com.regnosys.rosetta:pom:0.0.0.master from/to snapshots (https://regnosys.jfrog.io/regnosys/libs-snapshot): Access denied to https://regnosys.jfrog.io/regnosys/libs-snapshot/com/regnosys/rosetta/com.regnosys.rosetta/0.0.0.master/com.regnosys.rosetta-0.0.0.master.pom. Error code 403, Forbidden -> [Help 1]

This is because the POM of code generators specifies a dependancy on the latest unreleased version of the dsl project while only released versions are available externally

--- a/codefresh.yml
+++ b/codefresh.yml
@@ -30,7 +30,7 @@ steps:
       - mvn -s /settings.xml versions:set -DnewVersion=${{RELEASE_NAME}}
       - mvn -s /settings.xml versions:update-property -Dproperty=regnosys.rosetta.version -DnewVersion=[${{RELEASE_NAME}}] -DallowSnapshots=true -DallowDowngrade=true
       - mvn -s /settings.xml versions:update-property -Dproperty=regnosys.common.version -DnewVersion=[${{RELEASE_NAME}}] -DallowSnapshots=true -DallowDowngrade=true
-      - mvn -U -s /settings.xml clean deploy
+      - mvn -U -s /settings.xml -Pregnosys clean deploy
       
   TagRepo:
     title: Tag git repo with release name

--- a/daml/pom.xml
+++ b/daml/pom.xml
@@ -58,7 +58,6 @@
 		<dependency>
 			<groupId>com.regnosys</groupId>
 			<artifactId>rosetta-common</artifactId>
-			<version>${regnosys.common.version}</version>
 		</dependency>
 		<dependency>
 			<groupId>org.eclipse.emf</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -86,6 +86,11 @@
 				<version>${regnosys.rosetta.version}</version>
 			</dependency>
 			<dependency>
+				<groupId>com.regnosys</groupId>
+				<artifactId>rosetta-common</artifactId>
+				<version>${regnosys.common.version}</version>
+			</dependency>
+			<dependency>
 				<groupId>com.google.guava</groupId>
 				<artifactId>guava</artifactId>
 				<version>${guava.version}</version>

--- a/pom.xml
+++ b/pom.xml
@@ -1,18 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<!--
-Copyright (c) REGnosys 2019 (www.regnosys.com) All rights reserved.
+<!-- Copyright (c) REGnosys 2019 (www.regnosys.com) All rights reserved. 
+	This program is licensed to you under the Apache License Version 2.0, and 
+	you may not use this file except in compliance with the Apache License Version 
+	2.0. You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0. 
+	Unless required by applicable law or agreed to in writing, software distributed 
+	under the Apache License Version 2.0 is distributed on an "AS IS" BASIS, 
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+	See the Apache License Version 2.0 for the specific language governing permissions 
+	and limitations there under. -->
 
-This program is licensed to you under the Apache License Version 2.0,
-and you may not use this file except in compliance with the Apache License Version 2.0.
-You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
-Unless required by applicable law or agreed to in writing,
-software distributed under the Apache License Version 2.0 is distributed on an
-"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
--->
-
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
 	<modelVersion>4.0.0</modelVersion>
@@ -33,14 +32,32 @@ See the Apache License Version 2.0 for the specific language governing permissio
 		<url>https://www.regnosys.com/</url>
 	</organization>
 
+	<profiles>
+		<profile>
+			<id>regnosys</id>
+			<properties>
+				<regnosys.rosetta.version>0.0.0.master</regnosys.rosetta.version>
+				<regnosys.common.version>0.0.0.master</regnosys.common.version>
+			</properties>
+		</profile>
+		<profile>
+			<id>external</id>
+			<activation>
+				<activeByDefault>true</activeByDefault>
+			</activation>
+			<properties>
+				<regnosys.rosetta.version>2.21.1</regnosys.rosetta.version>
+				<regnosys.common.version>2.21.1</regnosys.common.version>
+			</properties>
+		</profile>
+	</profiles>
+
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<maven.compiler.source>1.8</maven.compiler.source>
 		<maven.compiler.target>1.8</maven.compiler.target>
 		<java.version>1.8</java.version>
 		<org.eclipse.emf.ecore.version>2.15.0</org.eclipse.emf.ecore.version>
-		<regnosys.rosetta.version>0.0.0.master</regnosys.rosetta.version>
-		<regnosys.common.version>0.0.0.master</regnosys.common.version>
 		<xtext.version>2.15.0</xtext.version>
 		<guava.version>21.0</guava.version>
 		<guice.version>3.0</guice.version>

